datasets/in/mha_banned/crawler.py

```diff
@@ -16,37 +16,53 @@ def parse_names(field: str) -> List[str]:
     return names
 
 
-def assert_link_hash(context: Context, doc: _Element, label: str, expected: str) -> str:
+def assert_link_hash(
+    context: Context,
+    doc: _Element,
+    label: str,
+    expected: str,
+    xpath: Optional[str] = None,
+) -> str:
     label_xpath = f".//td[contains(text(), '{label}')]"
     label_cells = doc.xpath(label_xpath)
     assert len(label_cells) == 1
     anchors = label_cells[0].xpath("./following-sibling::td//a")
     assert len(anchors) == 1
     link = anchors[0]
     url = link.get("href")
-    h.assert_url_hash(context, url, expected)
-    return url
+    if xpath:
+        success = h.assert_html_url_hash(context, url, expected, path=xpath)
+        if not success:
+            print(f"Hash assertion for {url} with XPath {xpath} failed")
+    else:
+        success = h.assert_url_hash(context, url, expected)
+        if not success:
+            print(f"Hash assertion for {url} failed")
+
+    return url if success else None
```

Comment:
we don't want to leave this as print for production. assert_... would be logging a warning for us if the hash doesn't match. 

we also always want to add the link we find on the overview page to the sources set.

we're doing two important things in this function:

- looking up the link to the document based on the label (where the label is a part of the legislation we care about)
- checking if the hash of that link's content has changed
I think that's making it a bit difficult to understand.

Perhaps now or later, we could split this into two functions:
- get_link_by_label()
- assert_link_hash()
That would also make it clearer why we return the url despite the hash not matching.

Perhaps renaming "expected_sources" to "linked_sources" would make more sense?